zenoh/src/api/liveliness.rs

```diff
@@ -578,11 +575,13 @@ where
         session
             .declare_liveliness_subscriber_inner(&key_expr, Locality::default(), callback)
             .map(|sub_state| Subscriber {
-                subscriber: SubscriberInner {
-                    session,
+                inner: SubscriberInner {
+                    #[cfg(feature = "unstable")]
+                    session_id: session.zid(),
+                    session: session.into(),
                     state: sub_state,
                     kind: SubscriberKind::LivelinessSubscriber,
-                    undeclare_on_drop: true,
+                    undeclare_on_drop: size_of::<Handler::Handler>() > 0,
```

Comment:
I understand that a zero-sized handler is not dropped. Callback are zero-size and therefore the handler will not be dropped. Could you add a comment explaining this?